src/hashes/blake2b.c

```diff
@@ -342,7 +342,7 @@ int blake2b_process(hash_state *md, const unsigned char *in, unsigned long inlen
       unsigned long fill = BLAKE2B_BLOCKBYTES - left;
       if (inlen > fill) {
          md->blake2b.curlen = 0;
-         XMEMCPY(md->blake2b.buf + left, in, fill); /* Fill buffer */
+         XMEMCPY(md->blake2b.buf + (left % sizeof(md->blake2b.buf)), in, fill); /* Fill buffer */
```

Comment:
I guess that we could somehow re-factor this entire part, but I didn't see an added value... therefore I went for the cheap trick to make the pointer always valid ;-)
If you think we shouldn't do it like that I'll investigate a bit further.